src/Umbraco.Web.UI.NetCore/Umbraco/PartialViewMacros/Templates/Gallery.cshtml

```diff
@@ -1,10 +1,14 @@
 @using Umbraco.Core.Models.PublishedContent
 @using Umbraco.Web
 @using Umbraco.Core
-@using Umbraco.Core.Media
-@using Umbraco.Web.Composing
-@inherits Umbraco.Web.Macros.PartialViewMacroPage
-
+@using Umbraco.Core.Composing
+@using Umbraco.Core.PropertyEditors.ValueConverters
+@using Umbraco.Web.Routing
+@inherits Umbraco.Web.Common.Macros.PartialViewMacroPage
+@inject IPublishedContentQuery PublishedContentQuery
+@inject IVariationContextAccessor VariationContextAccessor
+@inject IPublishedUrlProvider PublishedUrlProvider
+@inject IFactory Factory
```

Comment:
Why is Factory injected?  You can inject the `IImageUrlGenerator` instead

Reply:
Makes sense! But as I mentioned after we uncomment ln.[53](https://github.com/umbraco/Umbraco-CMS/blob/717d8e6156d49665a53d230608e38ee276e49c68/src/Umbraco.Web.UI.NetCore/Umbraco/PartialViewMacros/Templates/Gallery.cshtml#L53)-57, we will have problems with the `GetCropUrl`, so perhaps we need to introduce a new extension method or get the crop URL differently?